Emby.Server.Implementations/Library/MediaStreamSelector.cs

```diff
@@ -45,42 +45,42 @@ public static class MediaStreamSelector
                 .ThenByDescending(x => x.IsDefault)
+                .ThenByDescending(x => preferredLanguages.Contains(x.Language, StringComparison.OrdinalIgnoreCase))
                 .ToList();
 
             MediaStream? stream = null;
             if (mode == SubtitlePlaybackMode.Default)
             {
-                // Prefer embedded metadata over smart logic
-                stream = sortedStreams.FirstOrDefault(s => s.IsExternal || s.IsForced || s.IsDefault);
-
-                // if the audio language is not understood by the user, load their preferred subs, if there are any
-                if (stream == null && !preferredLanguages.Contains(audioTrackLanguage, StringComparison.OrdinalIgnoreCase))
-                {
-                    stream = sortedStreams.FirstOrDefault(s => !s.IsForced && preferredLanguages.Contains(s.Language, StringComparison.OrdinalIgnoreCase));
-                }
+                // Load subtitles according to external, forced and default flags.
+                stream = sortedStreams.FirstOrDefault(x => x.IsExternal || x.IsForced || x.IsDefault);
             }
             else if (mode == SubtitlePlaybackMode.Smart)
             {
-                // if the audio language is not understood by the user, load their preferred subs, if there are any
+                // Only attempt to load subtitles if the audio language is not one of the user's preferred subtitle languages.
+                // If no subtitles of preferred language available, use default behaviour.
                 if (!preferredLanguages.Contains(audioTrackLanguage, StringComparison.OrdinalIgnoreCase))
                 {
-                    stream = streams.FirstOrDefault(s => !s.IsForced && preferredLanguages.Contains(s.Language, StringComparison.OrdinalIgnoreCase)) ??
-                        streams.FirstOrDefault(s => preferredLanguages.Contains(s.Language, StringComparison.OrdinalIgnoreCase));
+                    stream = sortedStreams.FirstOrDefault(x => preferredLanguages.Contains(x.Language, StringComparison.OrdinalIgnoreCase)) ??
+                        sortedStreams.FirstOrDefault(x => x.IsExternal || x.IsForced || x.IsDefault);
+                }
+                else
+                {
+                    // Respect forced flag.
+                    stream = sortedStreams.FirstOrDefault(x => x.IsForced);
                 }
             }
             else if (mode == SubtitlePlaybackMode.Always)
             {
-                // always load the most suitable full subtitles
-                stream = sortedStreams.FirstOrDefault(s => !s.IsForced);
+                // Always load subtitles of the user's preferred subtitle language if possible, otherwise default behaviour.
+                stream = sortedStreams.FirstOrDefault(x => preferredLanguages.Contains(x.Language, StringComparison.OrdinalIgnoreCase)) ??
+                    sortedStreams.FirstOrDefault(x => x.IsExternal || x.IsForced || x.IsDefault);
```

Comment:
Forced tracks sort first and are limited to what needs to be translated to the language of the sub track, where mode `Always` means that the user wants subtitles for everything. I think this would benefit from the original `!s.IsForced` check, at least when comparing to `preferredLanguages`.

Reply:
It's a good point that the user would prefer full subtitles so I've re-added the check.

Do you think it's worth considering the case where an external track in an alternate language is chosen over a forced track in the desired language? I'm thinking that would generally be desired because downloading external tracks is nearly always done purposefully?

Reply:
Yeah, I think anything not forced and in `preferredLanguages` is a valid choice, the rest of the sorting and selecting looks reasonable to me.